Sources/Widgets/Widgets3D/LineWidget/behavior.js

```diff
@@ -99,7 +119,54 @@ export default function widgetBehavior(publicAPI, model) {
     );
   }
 
-  // set in public to update handle  Direction when handle change in UI
+  function orientFirstHandleBeforeSecondHandlePlacement(callData) {
+    const handle1Pos = model.widgetState.getHandle1().getOrigin();
+    const WorldMousePos = publicAPI.computeWorldToDisplay(
+      model.renderer,
+      handle1Pos[0],
+      handle1Pos[1],
+      handle1Pos[2]
+    );
+    const mousePos = publicAPI.computeDisplayToWorld(
+      model.renderer,
+      callData.position.x,
+      callData.position.y,
+      WorldMousePos[2]
+    );
+    vtkMath.subtract(
+      model.widgetState.getHandle1().getOrigin(),
+      mousePos,
+      Direction
+    );
+  }
+
+  function orientHandlesWithCompleteWidget(modifier) {
+    const handle1Pos = model.widgetState.getHandle1().getOrigin();
+    const handle2Pos = model.widgetState.getHandle2().getOrigin();
+    vtkMath.subtract(handle1Pos, handle2Pos, Direction);
+    vtkMath.multiplyScalar(Direction, modifier);
```

Comment:
you need a function that computes a direction between 2 points. one function that calls it with p1 and p2, and one function that calls it with p1 and mouse.

This is called code factorization (to not duplicate code)